automerge-backend/src/actor_states.rs

```diff
@@ -0,0 +1,95 @@
+use crate::error::AutomergeError;
+use crate::protocol::{ActorID, Change, Clock};
+use std::collections::HashMap;
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct ActorStates {
+    change_by_actor: HashMap<ActorID, Vec<Change>>,
+    deps_by_actor: HashMap<ActorID, Vec<Clock>>,
+    pub history: Vec<Change>,
+}
+
+impl ActorStates {
+    pub(crate) fn new() -> ActorStates {
+        ActorStates {
+            change_by_actor: HashMap::new(),
+            deps_by_actor: HashMap::new(),
+            history: Vec::new(),
+        }
+    }
+
+    pub fn is_concurrent(
+        &self,
+        actor_id1: &ActorID,
```

Comment:
I'm concerned that by splitting out the actor ID and the sequence number from the `OperationWithMetadata` it becomes much easier to accidentally pass those arguments in the wrong order.  We only ever use this method to compare `OperationWithMetadata` structs, is there a reason not to take them directly as arguments?

Reply:
Thats fair - the function seemed really general purpose and the OpWMetadata seemed very situational - but its a good point - i put it back